### ISSUE TITLE
Better UI auto scaling in different screen resolution

### ISSUE THREAD

**Issue by Nebulosa-Cat** [garethgeorge/backrest#168]
For comparison
this is the preview in README:
![image](https://github.com/garethgeorge/backrest/assets/30948446/6da346e0-a3ff-447d-ba84-0e1dbf8f4b38)

this is how the UI look in 4k 28inch monitor:
100%
![image](https://github.com/garethgeorge/backrest/assets/30948446/fe35b16b-8402-4d08-a316-bfff61247e4e)
110%
![image](https://github.com/garethgeorge/backrest/assets/30948446/ec933581-0742-47c7-b0a9-539a6653e1f0)
120%
![image](https://github.com/garethgeorge/backrest/assets/30948446/d96b366e-ffe8-4f86-8fdd-589c77ec7fce)

I think the UI needs to add custom sizing options, especially as modern devices have many resolutions

For example, common screen on rack 1280x960 (4:3), normal 1920x1080 (16:9), 2k, 4k, 5k (like mac studio or some LG screen for mac user), evem some 13inch laptop or 11inch iPad... etc

May need a better design of auto-scaling default font sizes to fit more devices in browser.

**Comment by garethgeorge** [comment]
Thanks for sharing the screenshots! 

Brainstorming a few options to improve scaling across a few form factors

 * Looking at backrest on mobile, I already have some support for a custom layout here that hides the far right panel in favor of a popup when you click on a backup row BUT the sidebar and text is quite small. I'll see if I can expand this a bit.
 * On desktop, I think the main problem seems to be that the layout isn't resizeable. I'll aim to make the content area draggable to resize the backup pane / tree view panes.

Is font size an issue? How do you envisage scaling being configured?

**Comment by Nebulosa-Cat** [comment]
> Thanks for sharing the screenshots!
> 
> Brainstorming a few options to improve scaling across a few form factors
> 
> * Looking at backrest on mobile, I already have some support for a custom layout here that hides the far right panel in favor of a popup when you click on a backup row BUT the sidebar and text is quite small. I'll see if I can expand this a bit.
> * On desktop, I think the main problem seems to be that the layout isn't resizeable. I'll aim to make the content area draggable to resize the backup pane / tree view panes.
> 
> Is font size an issue? How do you envisage scaling being configured?

In Desktop, for example, my screen is a 28inch 4k screen, the distance to me is 60cm, the text letter wide is slightly larger than 1mm when use 100% scale, when change to 120% it will much clearer (~2mm)

in my 11inch iPad Pro, it looks like this:
![IMG_2166](https://github.com/garethgeorge/backrest/assets/30948446/1740ef0b-db69-494a-b1f8-14014841517b)
The main problem is letter size too. But in fact, the problem is smaller than on the desktop because the screen is much closer to the eyes when using a tablet.

in my 12 mini (It’s a unusual device use case. I don’t think anyone would want to use such a small phone for service management. It’s for comparison only.):
![IMG_5006](https://github.com/garethgeorge/backrest/assets/30948446/7e7bb501-9365-4a18-9021-1892d0140295)
If need use on phone, it maybe need single column layout i think, the "Plans", "Repo" an "Setting" maybe need fold.
![IMG_5008](https://github.com/garethgeorge/backrest/assets/30948446/5e2a1e22-cbc4-4b75-a8af-ea6be98b90bf)
and the detail box maybe need fill almost screen to get clear view

**Comment by Snake883** [comment]
I have to squint to see the small text (or zoom), especially the text at the far right:

![image](https://github.com/user-attachments/assets/3f5687fd-87e8-4093-85a1-33d1ca03277a)

**Comment by Vyerni** [comment]
I personally would definitely love the ability to increase the font size.

**Comment by pretubedongit** [comment]
I think the webui is not responsive/adaptive to mobile browsers. It looks like this on my 6.67 inch Android screen.

It's such a good project. I don't agree that using a phone for repo admin is rare. The whole idea of adding a web GUI to restic is to avoid using SSH and type long commands while I am on the go. With a GUI we can quickly check the repo without a physical keyboard. I just hope it is mobile friendly.

![Image](https://github.com/user-attachments/assets/ebf5278e-6fa9-41e4-ba71-de18c69516ab)

**Comment by ascott18** [comment]
Some of the font sizes are unreasonably small. The grey text here is 8.4px. I would consider 12px to be the absolute minimum that *anything* should *ever* be in a web UI

![Image](https://github.com/user-attachments/assets/ed6ca0f0-cd7e-4d78-9941-f9adeccc39f1)

**Comment by garethgeorge** [comment]
Appreciate all of the interest in this and just commenting to add that I recognize it's a pain point, unfortunately the way I've written the WebUI so far is very desktop focused. 

Antd does provide a mobile UI package https://mobile.ant.design/ , I think a future project could be cleaning up a number of backrest's UIs, factoring out more shareable elements into components, and making these responsive.

I think the biggest pain point by far is the side nav does not work well on mobile. I'll need to think through what a better layout might be.

**Comment by parer55** [comment]
Hello,
Thanks for the amazing work, this is a new must-have for me. 
Just came here to say this, and that I also would love some mobile friendly screens. 
Just a thought. 
Keep up the good work and thanks again!

**Comment by tobiasgraeber** [comment]
Yes, please. I also suggest add a mobile, responsive design. Should be high priority. Also, please, make the sidebar/main panel individually resizable, add a toggle and optimized it also for mobile usage. Latest UI/Design changes are a step forward but still a lot from UI/Mobile/Design perspektive is missing.

Thank you
